### PR TITLE
Bluetooth: Controller: Fix missing Broadcast ISO HCI reset

### LINKED ISSUE #59099: Bluetooth: Controller: Cannot create BIG after HCI reset
**Describe the bug**
There seems to be some leakage that prevents the host from creating a new BIG after a HCI reset, and a power cycle on the board seems to be the only workaround. 

**To Reproduce**
on the nRF5340:
Using https://github.com/zephyrproject-rtos/zephyr/pull/58914 with the bt audio shell (compile with audio.conf) on the appcore, and the https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/bluetooth/hci_rpmsg sample with https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/bluetooth/hci_rpmsg/nrf5340_cpunet_iso_receive-bt_ll_sw_split.conf on the network core, do the following in the shell: 

```
bt init sync
bap init
bt adv-create nconn-nscan ext-adv name-ad
cap_initiator ac_12 16_2_1
bt adv-data discov
bt per-adv-param
bt per-adv-data
bap start_broadcast
bt per-adv on
bt adv-start
```

and then do a HCI reset (I did it by running the shell as a `native_posix` application and restarting the application, and do the same above commands again. First attempt works and results in a broadcast source. Second attempt fails at creating the big: 

```
uart:~$ bap start_broadcast
[00:00:02.270,000] <wrn> bt_hci_core: bt_hci_cmd_send_sync: opcode 0x2068 status 0x0c
[00:00:02.270,000] <dbg> bt_bap_broadcast_source: bt_bap_broadcast_source_start: Failed to create BIG: -5
Unable to start broadcast source: -5
```

**Expected behavior**
Expect to create a broadcast source after a HCI reset. 

**Impact**
Annoyance

**Logs and console output**
[btmon.txt](https://github.com/zephyrproject-rtos/zephyr/files/11706867/btmon.txt)

**Environment (please complete the following information):**
 - Commit SHA or Version used: 028e0a3c8d75fc5e2027f54dbb5a55aadbb1c490

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by cvinayak**
Fix missing Broadcast ISO implementation for HCI reset.

Fixes #59099.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

**Comment by cvinayak** [comment]
@Tronil May be you can give your review?

**Review by Tronil — APPROVED** [review]
Looks good to me, but @mtpr-ot is more familiar with this part of the code

**Review by Thalley — DISMISSED** [review]

**Review by kruithofa — DISMISSED** [review]

**Review by aescolar — DISMISSED** [review]
A relevant issue and 2 nits

File: tests/bsim/bluetooth/ll/bis/tests_scripts/broadcast_iso_ll_interface.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# Copyright 2020 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+source ${ZEPHYR_BASE}/tests/bsim/sh_common.source
+
+# Basic ISO broadcast test: a broadcaster transmits a BIS and a receiver listens
+# to the BIS.
+simulation_id="broadcast_iso_ll_interface"
+verbosity_level=2
+EXECUTE_TIMEOUT=30
```
[the remark this answers is not in the document]
Comment (cvinayak) [reply tests/bsim/bluetooth/ll/bis/tests_scripts/broadcast_iso_ll_interface.sh]: Removed

File: tests/bsim/bluetooth/ll/bis/tests_scripts/broadcast_iso_ll_interface.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+cd ${BSIM_OUT_PATH}/bin
+
+Execute ./bs_${BOARD}_tests_bsim_bluetooth_ll_bis_prj_conf_overlay-ll_interface_conf \
```
[the remark this answers is not in the document]
Comment (cvinayak) [reply tests/bsim/bluetooth/ll/bis/tests_scripts/broadcast_iso_ll_interface.sh]: Fixed

File: tests/bsim/bluetooth/ll/compile.sh [hunk 3]
```diff
@@ -29,7 +29,9 @@ app=tests/bsim/bluetooth/ll/conn conf_file=prj_split_privacy.conf compile
 app=tests/bsim/bluetooth/ll/conn conf_file=prj_split_low_lat.conf compile
 app=tests/bsim/bluetooth/ll/conn conf_file=prj_split_single_timer.conf compile
 
-app=tests/bsim/bluetooth/ll/bis compile
+app=tests/bsim/bluetooth/ll/bis conf_file=prj.conf compile
```
[the remark this answers is not in the document]
Comment (cvinayak) [reply tests/bsim/bluetooth/ll/compile.sh]: removed.

**Review by Thalley — DISMISSED** [review]

**Review by Thalley — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**